### ISSUE TITLE
About test-time-refinement (TTR)

### ISSUE THREAD

**Issue by heiwang1997** [nianticlabs/manydepth#27]
Hi @JamieWatson683, thank you for this very exciting project! May I ask you a question: Do you provide code for the test-time-refinement (TTR) as shown in the main table of the Results section? If so, how to use that for my own sequence?

**Comment by JamieWatson683** [comment]
Hi - thanks for your interest, and sorry for the delay in getting back to you!

No unfortunately we do not provide this code as it was a little messy, but it wouldn't be too hard to implement if you wanted to. Let me know if you give it a try and I can perhaps give some pointers if you need!

Thanks a lot.

**Comment by heiwang1997** [comment]
Thank you very much for the response. I will then try to implement it myself and see what I can get.
Just a quick question before I close - how many epochs did you finetune for each sample and what learning rate did you use?

**Comment by JamieWatson683** [comment]
Sorry for the delay - we followed the work by McCraith et al (https://arxiv.org/pdf/2004.05821.pdf) and trained only the depth + pose encoder. 

I believe we used an LR of 0.025 and trained for 50 steps. I hope that helps!

**Comment by heiwang1997** [comment]
Cool! That perfectly answers my question. Thanks again for the reply and hope you all the best.

**Comment by p-andra** [comment]
Hi @JamieWatson683! Thanks for manydepth!

I have some questions regarding the implementation of the test-time refinement. Help would be greatly appreciated. Here are my questions:

1. Do you freeze the bn layers in both encoders and decoders?
2. Do you set bn layers to train() or eval() during training for 50 iterations?
3. What bn momentum do you use?
4. What SGD momentum value do you use?
5. Do you use weight decay?
6. Do you apply test-time refinement only on non-static frames or on all the frames from the Eigen test set?

Thanks!
